### DISCUSSION TITLE
@tasks.loop(time=x) not working for me

### DISCUSSION THREAD

**Discussion by Nalanii** [Rapptz/discord.py#9547] [Q&A] [answered]
Hello! 

I have a very basic program that I'm trying to get up and running but for whatever reason, the time param for tasks.loop is not working for me. Here is my code:
```
import os
from dotenv import load_dotenv

import datetime
import discord
from discord.ext import tasks, commands
load_dotenv()
TOKEN = os.getenv('DISCORD_TOKEN')
intents = discord.Intents.default()

bot = commands.Bot(command_prefix='&', intents=intents)

goodNightTime = datetime.time(hour=19, minute=46) #Create the time on which the task should always run
print('goodNightTime: ', goodNightTime)

@tasks.loop(time=goodNightTime) #Create the task
async def Goodnight():
    print("BEGIN GOODNIGHT!!!!!!!!!!!")
    channel = bot.get_channel(522975347759775759)
    await channel.send("Good night! Make sure to go to sleep early, and get enough sleep!")
    print("Night Working")

@tasks.loop(minutes=1) #Create the task
async def Good_night_2():
    print("1 min interval")
    print('current time: ', datetime.datetime.now())
    channel = bot.get_channel(522975347759775759)
    await channel.send("1 min interval")


@bot.event
async def on_ready():
    if not Goodnight.is_running():
        Goodnight.start() #If the task is not already running, start it.
        print("Good night task started")
    if not Good_night_2.is_running():
        Good_night_2.start() #If the task is not already running, start it.
        print("Good night 2 task started")

bot.run(TOKEN)
```
Here's a screenshot of my output: 
![Screenshot 2023-08-29 194811](https://github.com/Rapptz/discord.py/assets/45048743/908d8cb7-4ba7-4677-84bf-e086fd0bfe06)

Overveiw:
* both tasks are started on_ready
* Good_night_2() prints every one minute and sends the message in the bot as expected
* Goodnight() never works! 
* The time I'm passing into the tasks.loop(time=x) [aka goodNightTime] is always past the current time (as seen in the screenshot with red underlines)

PLEASE HELP!!

**Comment by fretgfr** [comment]
The `time` given defaults to UTC for a timezone if you don't specify per the [documentation](https://discordpy.readthedocs.io/en/latest/ext/tasks/index.html?highlight=loop%20time#discord.ext.tasks.loop).

**Comment by Rapptz** [comment, marked as answer]
This is an issue with timezones. Naive times for the task extension are interpreted as UTC time, `datetime.datetime.now()` is a naive datetime representing local time. In order to make your datetime proper you need to make it an aware time for your local timezone. Using `astimezone()` is enough for that. e.g.

```py
local_tz = datetime.datetime.now().astimezone().tzinfo
goodNightTime = datetime.time(hour=19, minute=46, tzinfo=local_tz)
```

**Reply by Nalanii** [reply]
Thank you so much 😭 I tried setting the timezone manually using timezone from pytz but it wasn't working for me. This did the trick!!
